File: BungiiAutomation/Bungii.Test.Regression.Android.Integration/Pages/MenuPages/InvitePage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace Bungii.Test.Regression.Android.Integration.Pages.MenuPages
{
    class InvitePage
    {
        public InvitePage(IWebDriver driver)
        {
            PageFactory.InitElements(driver, this);
        }

        //------Invite Page Elements-----------------------------------------------------------------
        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@text='INVITE']")]
        public IWebElement Header_Invite { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/invite_referral_imageview")]
        public IWebElement Invite_Logo { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/invite_referral_title")]
        public IWebElement Invite_Title { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/invite_referral_description")]
        public IWebElement Invite_Text { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/invite_referral_code_textview")]
        public IWebElement Invite_Code { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/invite_referral_share_button")]
        public IWebElement Button_Share { get; set; }

        //------Share to---------------------------------------------------------------------------
        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@text='Share on Facebook']")]
        public IWebElement Share_Facebook { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@text='Share on Twitter']")]
        public IWebElement Share_Twitter { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@text='Share by Email']")]
        public IWebElement Share_Email { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@text='Share by Text Message']")]
        public IWebElement Share_TextMessage { get; set; }

        //------Share - Facebook App---------------------------------------------------------------
        [FindsBy(How = How.Id, Using = "com.facebook.katana:id/title_view")]
        public IWebElement FBApp_Title { get; set; }

        [FindsBy(How = How.Id, Using = "com.facebook.katana:id/status_text")]
        public IWebElement FBApp_StatusText { get; set; }

        [FindsBy(How = How.Id, Using = "com.facebook.katana:id/link_attachment_context_text")]
        public IWebElement FBApp_PreviewText { get; set; }

        [FindsBy(How = How.Id, Using = "com.facebook.katana:id/button_share")]
        public IWebElement FBApp_PostLink { get; set; }

        //------Share - Samsung Msg App-------------------------------------------------------------
        [FindsBy(How = How.Id, Using = "com.android.mms:id/edit_text_bottom")]
        public IWebElement Samsung_TextMsg_TextField { get; set; }

        //------Share - Gmail App-------------------------------------------------------------------
        [FindsBy(How = How.Id, Using = "com.google.android.gm:id/subject")]
        public IWebElement Gmail_Referral_Subject { get; set; }

        [FindsBy(How = How.Id, Using = "com.google.android.gm:id/body")]
        public IWebElement Gmail_Referral_Body { get; set; }

        //------Share - Twitter App-----------------------------------------------------------------
        [FindsBy(How = How.Id, Using = "status")]
        public IWebElement Twitter_Referral_Body { get; set; }
    }
}

File: Bungii.Test.Integration/src/test/java/com/bungii/android/pages/customer/InvitePage.java
package com.bungii.android.pages.customer;

import com.bungii.common.core.PageBase;
import org.openqa.selenium.WebElement;

public class InvitePage extends PageBase {

    //------Invite Page Elements-----------------------------------------------------------------
    public WebElement Header_Invite() { return findElement("//android.widget.TextView[@text='INVITE']", LocatorType.XPath); }

    public WebElement Invite_Logo() { return findElement("com.bungii.customer:id/invite_referral_imageview", LocatorType.Id); }

    public WebElement Invite_Title() { return findElement("com.bungii.customer:id/invite_referral_title", LocatorType.Id); }

    public WebElement Invite_Text() { return findElement("com.bungii.customer:id/invite_referral_description", LocatorType.Id); }

    public WebElement Invite_Code() { return findElement("com.bungii.customer:id/invite_referral_code_textview", LocatorType.Id); }

    public WebElement Button_Share() { return findElement("com.bungii.customer:id/invite_referral_share_button", LocatorType.Id); }

    //------Share to---------------------------------------------------------------------------
    public WebElement Share_Facebook() { return findElement("//android.widget.TextView[@text='Share on Facebook']", LocatorType.XPath); }

    public WebElement Share_Twitter() { return findElement("//android.widget.TextView[@text='Share on Twitter']", LocatorType.XPath); }

    public WebElement Share_Email() { return findElement("//android.widget.TextView[@text='Share by Email']", LocatorType.XPath); }

    public WebElement Share_TextMessage() { return findElement("//android.widget.TextView[@text='Share by Text Message']", LocatorType.XPath); }

    //------Share - Facebook App---------------------------------------------------------------
    public WebElement FBApp_Title() { return findElement("com.facebook.katana:id/title_view", LocatorType.Id); }

   // public WebElement FBApp_StatusText() { return findElement("com.facebook.katana:id/status_text", LocatorType.Id); }
   public WebElement FBApp_StatusText() { return findElement("//android.widget.EditText[@resource-id='com.facebook.katana:id/(name removed)']", LocatorType.XPath); }

    public WebElement FBApp_PreviewText() { return findElement("com.facebook.katana:id/link_attachment_context_text", LocatorType.Id); }

  //  public WebElement FBApp_PostLink() { return findElement("com.facebook.katana:id/button_share", LocatorType.Id); }

    public WebElement FBApp_PostLink(boolean ...ignoreException) { return findElement("//android.widget.Button[@content-desc=\"POST\"]", LocatorType.XPath,ignoreException); }
  //  public WebElement FBApp_PostLink(boolean ...ignoreException) { return findElement("//android.widget.ImageView[@content-desc=\"Set Album\"]/following-sibling::android.widget.TextView", LocatorType.XPath,ignoreException); }


    //------Share - Samsung Msg App-------------------------------------------------------------
    public WebElement Samsung_TextMsg_TextField() { return findElement("com.android.mms:id/edit_text_bottom", LocatorType.Id); }

    public WebElement TextMsg_TextField() { return findElement("com.android.mms:id/embedded_text_editor", LocatorType.Id); }

    //------Share - Gmail App-------------------------------------------------------------------
    public WebElement Gmail_Referral_Subject() { return findElement("com.google.android.gm:id/subject", LocatorType.Id); }

    public WebElement Gmail_Referral_Body() { return findElement("//*[@resource-id='com.google.android.gm:id/body_wrapper']/descendant::android.view.View", LocatorType.XPath); }

    //------Share - Twitter App-----------------------------------------------------------------
    public WebElement Twitter_Referral_Body(boolean ...ignoreException) { return findElement("//*[@resource-id='status']", LocatorType.XPath,ignoreException); }
    public WebElement Twitter_Referral_BodyChrome() { return findElement("android.widget.EditText", LocatorType.ClassName); }




    public WebElement Button_Back(){return findElement("//android.widget.ImageButton[@content-desc=\"Navigate up\"]",LocatorType.XPath);}

}
